process/interface.go

```diff
@@ -18,16 +18,16 @@ type ShardCoordinator interface {
 	IsInterfaceNil() bool
 }
 
-// ExtendedHeaderHandler defines what a subscribed handler to SovereignNotifier should do
-type ExtendedHeaderHandler interface {
-	ReceivedExtendedHeader(header *block.ShardHeaderExtended)
+// HeaderSubscriber defines a subscribed to incoming headers
+type HeaderSubscriber interface {
+	AddHeader(headerHash []byte, header data.HeaderHandler)
```

Comment:
ReceivedHeader ?

Reply:
+1

Reply:
No, this should be compatible with `AddHeader` from `HeadersPool` in `mx-chain-go`